rkyv/src/impls/smolstr.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{
+    ser::{ScratchSpace, Serializer},
+    string::{ArchivedString, StringResolver},
+    Archive, Deserialize, Fallible, Serialize,
+};
+use smol_str::SmolStr;
+
+impl Archive for SmolStr {
+    type Archived = ArchivedString;
+    type Resolver = StringResolver;
+
+    #[inline]
+    unsafe fn resolve(&self, pos: usize, resolver: Self::Resolver, out: *mut Self::Archived) {
+        ArchivedString::resolve_from_str(self, pos, resolver, out);
+    }
+}
+
+impl<S: ScratchSpace + Serializer + ?Sized> Serialize<S> for SmolStr {
+    #[inline]
+    fn serialize(&self, serializer: &mut S) -> Result<Self::Resolver, S::Error> {
+        ArchivedString::serialize_from_str(self, serializer)
+    }
+}
+
+impl<D: Fallible + ?Sized> Deserialize<SmolStr, D> for ArchivedString {
+    #[inline]
+    fn deserialize(&self, _deserializer: &mut D) -> Result<SmolStr, D::Error> {
+        Ok(if self.as_str().len() <= 23 {
```

Comment:
It looks like `SmolStr::new` will do this inline/heap-allocated check for us, so we can just call `SmolStr::new(self.as_str())` here in both cases.